### PR TITLE
UFS-coastal: CICE implementation Open boundary conditions

### PR DESCRIPTION
**Pull request by SmithJos13**
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>

There is a small tweak to the error handling in icepack. For small ice concentration, the model will sometimes crash. There is error checking now that ignores the error if the ice is less than 5cm thick. Otherwise, the model will crash if the error is thrown and the ice is thicker than 5cm.

## PR checklist
- [ ] Short (1 sentence) summary of your PR: 
    Boundary conditions
- [ ] Developer(s): 
    Joseph C. Smith
- [ ] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    ENTER INFORMATION HERE 
- How much do the PR code changes differ from the unmodified code? 
    - [ ] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on CICE or any other models?
    - [ ] Yes
    - [ ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/.)
    - [ ] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [ ] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.

### CODE REVIEW

File: columnphysics/icepack_therm_vertical.F90 [hunk 1]
```diff
@@ -2731,10 +2731,21 @@ subroutine icepack_step_therm1(dt,                        &
                                  prescribed_ice=prescribed_ice)
 
             if (icepack_warnings_aborted(subname)) then
-               write(warnstr,*) subname, ' ice: Vertical thermo error, cat ', n
-               call icepack_warnings_add(warnstr)
-               return
+               
+               if ((vicen(n)/aicen(n)) > real(5e-2)) then
```
Comment (uturuncoglu) [inline columnphysics/icepack_therm_vertical.F90]: @SmithJos13 I am not sure how you end up `5e-2` ratio but do you think tat it is applicable for all the cases. I also wonder if this change is really needed in this level. At this point, the main aim is to keep the changes minimal and isolated. So, any other configuration needs to work as it is but your changes will allow to run your case.

File: columnphysics/icepack_therm_vertical.F90 [hunk 2]
```diff
@@ -2731,10 +2731,21 @@ subroutine icepack_step_therm1(dt,                        &
...
+                  write(warnstr,*) subname, ' ice: Vertical thermo error, cat ', n
+                  call icepack_warnings_add(warnstr)
+                  return
+               else 
+                  call icepack_warnings_setabort(.false.,__FILE__,__LINE__)
+               endif
+               
+               !write(warnstr,*) subname, ' ice: Vertical thermo error, cat ', n
```
Comment (uturuncoglu) [inline columnphysics/icepack_therm_vertical.F90]: @SmithJos13 Please remove any unused code to make it more clear.

**Review by uturuncoglu — CHANGES_REQUESTED** [review]

**Comment by SmithJos13** [comment]
I have added some code that isolates this change based on whether the boundary condition option is active. If the option is not active, then it skips this portion of the code and it should function as intended.  

I have added the code to passage the variable names from CICE to icepack. Which is why there are two files changed now.

**Review by uturuncoglu — APPROVED** [review]
